### PR TITLE
Sorting Tool

### PR DESCRIPTION
**Pull request by TechSpiritSS**
## Type of Pull Request

- [x] Bug fix
- [ ] Enhancement

Related Issue #s or links (if any): 
#21 
## Description of Changes
Everything is working great and tested except for one bug. 
User is required to click sort twice in case of New Line separator to get the desired result. I'm not sure why this is happening but I wasn't able to resolve this bug.

### CODE REVIEW

**Comment by dlford** [comment]
@TechSpiritSS I'll look into that issue and get back to you, we can't merge as is unfortunately.

**Comment by TechSpiritSS** [comment]
:') well, that's fine, if you know the reason for the bug do let me know as well.

File: pages/sorting.tsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+              if (a.length === b.length) {
+                return b.localeCompare(a);
+              }
+              return b.length - a.length;
+            }
+            return 0;
+          });
+
+    setOutputText(outputTextArray.join(separator));
+    if (separator === '\n') setSeparator('lineBreak');
```
Comment (dlford) [inline pages/sorting.tsx:115]: Here is the cause of the issue requiring line break to be sorted twice. I think it'd be  best to store `separator` as an object with title and filter values, something like this:

```js
const separators = [
  { title: 'Line Break', filter: '\n' },
  // ...
];
```

That way we can keep the logic separate from the displayed text, and not have to do any juggling of that value to make things work.

**Review by dlford — CHANGES_REQUESTED** [review]

File: pages/sorting.tsx (hunk 1 above)
Comment (TechSpiritSS) [reply pages/sorting.tsx:115]: Let me try to fix this.

Comment (TechSpiritSS) [reply pages/sorting.tsx]: Resolved the issue, seems like I just needed a little break from the code to fix it.

**Review by dlford — CHANGES_REQUESTED** [review]
@TechSpiritSS Awesome! Code looks solid, just a few UI tweaks.

ksnip_20221008-101144

ksnip_20221008-101317

Also, I think it would add a lot of utility if we could add a field for the output separator so you could change a newline separated list to comma separated, etc. (This would also require a sort option of "keep order").

If that's too much to ask I can create a new issue for it.

**Comment by TechSpiritSS** [comment]
I think it's better to change that into new issue as I might not be able to commit those changes and I am travelling for next two days.

**Comment by dlford** [comment]
> I think it's better to change that into new issue as I might not be able to commit those changes as I am travelling for next two days.

No worries, just the UI fixes then please

**Comment by dlford** [comment]
@TechSpiritSS are you still wanting to work on this?

**Comment by TechSpiritSS** [comment]
Yes, I'll work on this, got some college stuff to deal with this week.

**Comment by dlford** [comment]
No worries, thank you!

**Pull request closed without merging**